graphql/schemabuilder/types.go

```diff
@@ -263,6 +267,29 @@ func (s *Object) ManualPaginationWithFallback(name string, manualPaginatedFunc i
 	s.Methods[name] = m
 }
 
+func (s *Object) FederatedFieldFunc(name string, f interface{}, options ...FieldFuncOption) {
+
+	if s.Methods == nil {
+		s.Methods = make(Methods)
+	}
+
+	m := &method{Fn: f}
+	for _, opt := range options {
+		opt.apply(m)
+	}
+
+	if _, ok := s.Methods[name]; ok {
+		panic("duplicate method")
+	}
+	federatedMethodName := fmt.Sprintf("%s-%s", name, s.ServiceName)
```

Comment:
How do you feel to switch name and s.ServiceName? 
It is more natural to me to have the serviceName first.

Reply:
I think I slightly prefered the object name in front so that when I look at all the federated methods in the format of "<object>-<serviceName>", I can think off all the services that know about that object together. Its also an internal format that isn't surface, so I can switch it if you strongly prefer the other way.